### ISSUE TITLE
Nested route doesn't work with nested routes using zod-openapi

### ISSUE THREAD

**Issue by mandado** [honojs/middleware#312]
Good afternoon everyone,

I'm having a problem with group routes in hono/openapi. When I use multiple parameters, the parameters are not replaced correctly in the swagger. The {nestedId} parameter is replaced with :nestedId instead. 

This seems like a bug. Is there a workaround for that for now ?

### Screenshot Documentation:
![image](https://github.com/honojs/middleware/assets/2823627/1186287c-d59c-4aba-bee5-fb063964e742)
### Code to reproduce:
```typescript
import { handle } from 'hono/aws-lambda'
import { swaggerUI } from "@hono/swagger-ui";
import { OpenAPIHono, createRoute, z } from "@hono/zod-openapi";

const app = new OpenAPIHono()

app.get(
  "/ui",
  swaggerUI({
    url: `/doc`
  }),
);

app.doc('/doc', c => ({
  openapi: '3.0.0',
  info: {
    version: '1.0.0',
    title: 'my API',
  },
  servers: [
    {
      url: `http://localhost:3000/`,
      description: 'Current environment',
    },
  ],
}))

const nested = new OpenAPIHono();
nested.openapi(createRoute({
  method: 'get',
  path: '/nested/{id}',
  responses: {
    200: {
      description: 'Nested response',
      content: {
        'application/json': {
          schema: z.object({
            nested: z.boolean(),
          })
        }
      }
    }
  }
}), (c) => c.json({ nested: true }));

app.route('/nested/:nestedId', nested);

export const handler = handle(app)
```

**Comment by mandado** [comment]
I think this happen since on that line https://github.com/honojs/middleware/blob/main/packages/zod-openapi/src/index.ts#L373 just merge but not transform the path to be compatible with the swagger path route. 

Might the solution for that is turn `/nested/:nestedId` to `/nested/{nestedId}` on that line https://github.com/honojs/middleware/blob/main/packages/zod-openapi/src/index.ts#L373 before merge.

**Comment by yusukebe** [comment]
Hi @mandado 

This issue might be fixed by #316. Try it out!
